src/utils/utils-vega.js

```diff
@@ -193,3 +194,448 @@ export function createRasterLayerGetterSetter(
+  // These will be used to properly style the svg popup object
+  const rndrProps = {}
+  if (
+    _vega &&
+    Array.isArray(_vega.marks) &&
+    _vega.marks.length > 0 &&
+    _vega.marks[0].properties
+  ) {
+    const propObj = _vega.marks[0].properties
+
+    renderAttributes.forEach(prop => {
+      if (
+        typeof propObj[prop] === "object" &&
+        propObj[prop].field &&
+        typeof propObj[prop].field === "string"
+      ) {
+        rndrProps[prop] = propObj[prop].field
+      }
+    })
+  }
+
+  // If the poly we hit-test is small, we'll scale it so that it
+  // can be seen. The minPopupArea is the minimum area of the popup
+  // poly, so if the poly's bounds is < minPopupArea, we'll scale it
+  // up to that size.
+  let scale = 1
+  const scaleRatio = minPopupArea / AABox2d.area(bounds)
+  const isScaled = scaleRatio > 1
+  if (isScaled) {
+    scale = Math.sqrt(scaleRatio)
+  }
+
+  // Now grab the style properties for the svg calculated from the vega
+  const popupStyle = _layer.popupStyle()
+  let fillColor = _layer.getFillColorVal(data[rndrProps.fillColor])
+  let strokeColor = _layer.getStrokeColorVal(data[rndrProps.strokeColor])
+  let strokeWidth = 1
+  if (typeof popupStyle === "object" && !isScaled) {
+    fillColor = popupStyle.fillColor || fillColor
+    strokeColor = popupStyle.strokeColor || strokeColor
+    strokeWidth = popupStyle.strokeWidth
+  }
+
+  // build out the svg
+  const svg = parentElem
+    .append("svg")
+    .attr("width", width)
+    .attr("height", height)
+
+  // transform svg node. This node will position the svg appropriately. Need
+  // to offset according to the scale above (scale >= 1)
+  const boundsCtr = AABox2d.getCenter(Point2d.create(), bounds)
+  const layerType = _layer.layerType()
+  const xform = svg
+    .append("g")
+    .attr("class", layerType === "polys" ? "map-poly-xform" : "map-polyline")
+    .attr(
+      "transform",
+      "translate(" +
+      (scale * bounds[AABox2d.MINX] - (scale - 1) * boundsCtr[0]) +
+      ", " +
+      (scale * (bounds[AABox2d.MINY] + 1) -
+        (scale - 1) * (boundsCtr[1] + 1)) +
+      ")"
+    )
+
+  // now add a transform node that will be used to apply animated scales to
+  // We want the animation to happen from the center of the bounds, so we
+  // place the transform origin there.
+  const group = xform
+    .append("g")
+    .attr("class", layerType === "polys" ? "map-poly" : "map-polyline")
+    .attr("transform-origin", `${boundsSz[0] / 2} ${boundsSz[1] / 2}`)
+
+  // inherited animation classes from css
+  if (animate) {
+    if (isScaled) {
+      group.classed("popupPoly", true)
+    } else {
+      group.classed("fadeInPoly", true)
+    }
+  }
+
+  // now apply the styles
+  if (typeof strokeWidth === "number") {
+    group.style("stroke-width", strokeWidth)
+  }
+
+  // applying shadow
+  const defs = group
+    .append('defs')
+
+  const filter = defs.append("filter")
```

Comment:
does this mean that poly pops now have drop shadows too? I'm wondering if it might be good to apply a style config to this function to turn on/off drop shadows, animation, etc. Or just handle this styling in CSS.

Reply:
yes it was, but I removed it from poly.